### PR TITLE
MM-47103 : Migrate components/channel_notifications_modal/components/collapse_view.jsx to Typescript

### PR DESCRIPTION
**Pull request by VishakhaPoonia**
#### Summary
Migrate components/channel_notifications_modal/components/collapse_view.jsx to Typescript

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/21066
Jira: https://mattermost.atlassian.net/browse/MM-47103
#### Related Pull Requests
NA

#### Screenshots
NA

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

**Review by AshishDhama — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
@VishakhaPoonia can you please check the CI failures

File: components/channel_notifications_modal/components/notification_section.jsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import React from 'react';
 
 import {NotificationSections, NotificationLevels} from 'utils/constants';
 
-import CollapseView from './collapse_view.jsx';
+import CollapseView from './collapse_view.tsx';
```
Comment (M-ZubairAhmed) [inline components/channel_notifications_modal/components/notification_section.jsx:9]: sorry i missed this we dont need the extension here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/channel_notifications_modal/components/notification_section.jsx (hunk 1 above)
Comment (VishakhaPoonia) [reply components/channel_notifications_modal/components/notification_section.jsx]: @M-ZubairAhmed Done

**Comment by M-ZubairAhmed** [comment]
https://github.com/mattermost/mattermost-webapp/pull/11274#issuecomment-1271341265

QA report still stands , no significant changes after this commit that would effect the tests

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @VishakhaPoonia 
Tested, looks good to merge.
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/248187-8bda03a-onprem-ent-webapp-pr-11274-mm-ee-test:8bda03a/mochawesome.html)- no PR related failures.

**Pull request merged**